lib/IRGen/GenClangDecl.cpp

```diff
@@ -82,6 +84,100 @@ clang::Decl *getDeclWithExecutableCode(clang::Decl *decl) {
   return nullptr;
 }
 
+auto walkCallGraphFromCtor(const clang::CXXConstructorDecl *toplevelCtor,
```

Comment:
It's too bad that the existing `ClangDeclFinder` can't be used/extended here. I'm going to look into this API a bit more to see if we could potentially re-use anything.